fsspec/implementations/memory.py

```diff
@@ -13,15 +13,26 @@
 class MemoryFileSystem(AbstractFileSystem):
     """A filesystem based on a dict of BytesIO objects
 
-    This is a global filesystem so instances of this class all point to the same
-    in memory filesystem.
+    Parameters
+    ----------
+    global_store : bool (True)
+        Whether to use global store, so instances of this class all point to
+        the same in memory filesystem. If False, every instance will have its
+        own store.
     """
 
-    store = {}  # global, do not overwrite!
-    pseudo_dirs = [""]  # global, do not overwrite!
+    store = {}  # global
+    pseudo_dirs = [""]  # global
     protocol = "memory"
     root_marker = "/"
 
+    def __init__(self, *args, **storage_options):
+        super().__init__(*args, **storage_options)
+        if not storage_options.get("global_store", True):
+            # detach from global
+            self.store = {}
+            self.pseudo_dirs = [""]
```

Comment:
It is interesting that this new fs "creation" is pretty different from what we do usually with filesystems. Reference fs seems similar, because you pass it the layout that it should be created from. Here we could also do smth like `MemoryFileSystem(store={})`, but that seems to be out of scope.